### PR TITLE
Revise SDL-0186 Template Titles

### PR DESCRIPTION
**Pull request by joeljfischer**
## Introduction
Update SDL-0186 Template Titles capability improvements by adding a new `TextFieldName` value and permitting the `templateTitle` to be altered by setting it to an empty string.

## Motivation
The design currently provides no support for developers to know the length of the text field available to them. Second, the design should permit removing the title by setting an empty string.

## Proposed solution
The proposed changes are as follows:

```xml
<enum name="TextFieldName" since="1.0">
    <!-- Existing RPCs -->

    <!-- Additions -->
    <element name="templateTitle" since="x.x">
        <description>The template title field in Show</description>
    </element>
</enum>
```

and

```xml
<param name="templateTitle" type="String" minlength="0" maxlength="100" mandatory="false">
```

where the `minlength` has been added.

## Potential downsides
No downsides identified; this is a very small update.

## Impact on existing code
The implementations will have to be updated

## Alternatives considered
No alternatives considered

### CODE REVIEW

**Comment by joeljfischer** [comment]
@theresalech this is ready for review

File: proposals/0186-template-titles.md [hunk 1]
```diff
@@ -30,6 +30,15 @@ Basic templates have nothing to orient the user to where they are in the UI. A t
         </description>
     </param>
 </function>
+
+<enum name="TextFieldName" since="1.0">
+    <!-- Existing RPCs -->
+
+    <!-- Additions -->
+    <element name="templateTitle" since="x.x">
+        <description>The template title field in Show</description>
```
Comment (ShobhitAd) [inline proposals/0186-template-titles.md]: Might be better to change `in Show` to `; applies to Show` to make it more consistent with the other elements in `TextFieldName`

Comment (joeljfischer) [reply proposals/0186-template-titles.md]: Fixed

**Comment by joeljfischer** [comment]
@theresalech I have updated the PR with an additional change requested by the Core team.

**Pull request merged**